### PR TITLE
Adding Stem Stage

### PR DESCRIPTION
**Pull request by sayannath**
Added the Conv Stem Stage and also wrote the configs for all the variants of models. 

The Output shape of the Conv stem stage is (112, 112, 64) which is same as the shape reported in the [paper](https://arxiv.org/pdf/2106.04803.pdf).

```
print(conv_2.shape)
```
This is written for verification purpose.

- [X] Stem Block
- [ ] MBConv Block
- [ ] Relative Attention Block
- [ ] FFN Block
- [ ] Output Block

### CODE REVIEW

File: coatnet/models/blocks.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import tensorflow as tf
+import tensorflow_addons as tfa
+from tensorflow.keras import layers
+
+
+def conv_3x3(input_layer, num_filters=64, downsample=False):
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: Prefer adding detailed docstrings and type annotations.

File: coatnet/models/blocks.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    stride = 1 if downsample == False else 2
+    conv_1 = layers.Conv2D(
+        filters=num_filters,
+        kernel_size=(3, 3),
+        strides=stride,
+        padding="same",
+        use_bias=False,
+    )(input_layer)
+    bn_1 = layers.BatchNormalization()(conv_1)
+    act_1 = tfa.layers.GELU()(bn_1)
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: `tf.nn.gelu`.

File: coatnet/models/coatnet.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        configs = get_model_config(self.model_name)
+
+        input_layer = keras.Input(shape=self.image_shape)  # Input Layer
+        conv_1 = conv_3x3(
+            input_layer=input_layer,
+            num_filters=configs.out_channels[0],
+            downsample=True,
+        )  # Conv Stem Block
+        conv_2 = conv_3x3(input_layer=conv_1, num_filters=configs.out_channels[0])
+        print(conv_2.shape)
```
Comment (sayakpaul) [inline coatnet/models/coatnet.py]: Any reason to implement it as a class rather than a method?

**Review by sayakpaul — CHANGES_REQUESTED** [review]

File: coatnet/models/coatnet.py (hunk 3 above)
Comment (sayannath) [reply coatnet/models/coatnet.py]: Nothing as such. Should I remove the class and rather use a function?

Comment (sayakpaul) [reply coatnet/models/coatnet.py]: Preferable.

Comment (sayakpaul) [reply coatnet/models/coatnet.py]: Also, please enlist the blocks that are yet to be implemented in the PR. 

Tick them off one by one as needed.

**Comment by sayannath** [comment]
@sayakpaul I have incorporated the changes.

**Comment by sayannath** [comment]
@sayakpaul any comments on the changes.

**Comment by sayakpaul** [comment]
Looks good.

**Pull request merged**